src/Nethermind/Nethermind.Blockchain/Receipts/ReceiptsRecovery.cs

```diff
@@ -97,12 +97,17 @@ public void RecoverReceiptData(TxReceipt receipt)
 
                 Transaction transaction = _block.GetNextTransaction();
 
+                if (transaction.SenderAddress is null && _forceRecoverSender)
+                {
+                    transaction.SenderAddress = _ecdsa.RecoverAddress(transaction, !_releaseSpec.ValidateChainId);
+                }
+
```

Comment:
`transaction.SenderAddress` is used by both `receipt.Sender` and `receipt.ContractAddress`, so the recovery step must be done before usage.

Reply:
Maybe add a simple test?

Reply:
Added test

Reply:
Added `TryRecover_should_recover_contract_address` test